vm/address/src/lib.rs

```diff
@@ -303,7 +304,69 @@ pub(crate) fn from_leb_bytes(bz: &[u8]) -> Result<u64, Error> {
     let mut readable = bz;
 
     // write id to buffer in leb128 format
-    Ok(leb128::read::unsigned(&mut readable)?)
+    let id = leb128::read::unsigned(&mut readable)?;
+
+    if to_leb_bytes(id)? == bz {
+        Ok(id)
+    } else {
+        error!(
```

Comment:
Probably makes sense to have this be part of the InvalidAddressIDPayload variant instead of printing it out

Reply:
Like, maybe using a tuple in the enum variant to capture the id?

Reply:
Take a look at these changes. Is this what you meant?
https://github.com/ChainSafe/forest/pull/1149/commits/2e5f084c5628bff42be67441aae92d563065fcb3